jiant/models.py

```diff
@@ -687,7 +690,33 @@ def build_single_sentence_module(task, d_inp: int, project_before_pooling: bool,
     return module
 
 
-def build_pair_sentence_module(task, d_inp, model, params):
+def build_sop(task, d_inp, model, params):
+    """
+    Build and load the pretrained head for the sentence order prediction task.
+    Parameters
+    ----------
+    task: Task,
+    d_inp: int,
+    model: MultiTaskModel,
+    params: Params
+
+    Returns
+    -------
+    module: Pooler object, which is loaded with pretrained weights from ALBERT SOP
+    pretraining. 
+
+    """
+    # By default, we do not project before pooling. Since the projection weights is what is
+    # exposed in huggingface for the pretrained weights, we set projection to true.
```

Comment:
Clarify: Is this a projection _before pooling_ , or something else? I'm not aware of any kind of pooling operation in ALBERT.

Reply:
What they call a "pooler" is a projection: 
https://github.com/huggingface/transformers/blob/ddb6f9476b58ed9bf4433622ca9aa49932929bc0/src/transformers/modeling_albert.py#L446

Reply:
and this is used here to create the pooled_output (which is the head used in the huggingface code for SOP prediction) https://github.com/huggingface/transformers/blob/ddb6f9476b58ed9bf4433622ca9aa49932929bc0/src/transformers/modeling_albert.py#L570

Reply:
basically, what the huggingface implementation of ALBERT does is, for the SOP head, takes the output of the ALBERT encoding and passes it through a linear layer and an activation function. The closest thing we have to that is this project_before_pooling attribute in the PairClassifier.

Reply:
Ah, got it. This could use a clarification in the code comments here, though it doesn't help that I've lost track of where our pooling logic lives.